docs/architecture/architecture.md

```diff
@@ -0,0 +1,330 @@
+<a name="overview"></a>
+
+# Architecture
+
+-   [Components](#Components)
+-   [Concepts](#Concepts)
+-   [Data flow](#Dataflow)
+-   [Scenarios](#Scenarios)
+
+<a name="Components"></a>
+
+## Components
+
+### perseo
+
+The CEP "front-end", it is in charge of processing incoming events and rules, storing rules and executing actions.
+Execution of actions are recorded for controlling frecuency of the actions performed. Also, it checks the no-update
+("no-signal") rules which set a maximum time interval between events from a context broker entity.
+
+### core (perseo-core)
+
+The CEP "back-end", the "rule-engine". It checks incoming events against rules in EPL and invokes perseo if an action
+must be executed. It has no pesistent storage. Rules are kept in memory. The whole set of rules is refreshed
+periodically by perseo "FE".
+
+### mongoDB
+
+Database used by perseo for storing rules and executions of actions.
+
+![Imagen](../images/components.png)
+
+## External components that interact with Perseo
+
+### Orion (Context Broker)
+
+Source of events from the entities it manages. Also, some actions perform an update of the entity that fired the action.
+The Context Broker as a target of actions can be secured with a PEP proxy protecting
+
+### Portal
+
+From point of view of CEP, the graphical user interface for creating rules (visual rules). It could be any user of the
+rules API.
+
+### Orion Database
+
+Perseo queries the Orion database periodically in order to check if the entities referred by any "non-update"
+("no-signal") rule should trigger an action in case these entities have been "silent" for too long.
+
+_NOTE:_ this is an interim mechanism in version 0.4.x. In the future, probably this will be done using the Orion API, so
+Perseo wouldn't interact any longer with the Orion database directly.
+
+### SMS gateway
+
+Some actions send an SMS to a number set in the action parameters. This is done by an HTTP post to the SMS gateway (SMPP
+adapter), configured in perseo.
+
+### SMPP server
+
+Some actions send an SMS to a number set in the action parameters. Alternativaly instead of use a SMS gateway this can
+be done by and HTTP post to a SMPP server configured in perseo.
+
+### SMTP server
+
+Some actions send an email. It is done using an SMTP server configured in perseo.
+
+### Generic HTTP server
+
+Some actions consist on making an HTTP POST to an URL provided as parameter of the action. his URL can point to any
+host.
+
+### Authorization server
+
+The interaction with a Context Broker through a PEP proxy requires an access token that must be refreshed periodically.
+The "trust token" associated with a rule that executes an update-action must be exchanged by an "access token" at the
+Authorization Server, when the access token has expired.
+
+<a name="Concepts"></a>
```

Comment:
Pointless anchor - it exists in the header below

Reply:
Ok. Fixed in ef4582b